### PR TITLE
feat(completion): add `nvim-cmp-buffer-lines`

### PR DESCRIPTION
**Pull request by Axlefublr**
## 📑 Description

Adds the `nvim-cmp-buffer-lines` nvim-cmp source, but instead of just adding it to normal completions, keeps it local to a hotkey.

The reasoning for this is that, even with a low priority, whole line completions end up being super annoying, when mixed in with all the other (usually more useful) suggestions. So it's for this reason that this source is effectively "opt-in".

The mapping used is `<C-x><C-l>`, because that's the default way to do whole line completions. What you basically get with this community pack, is you replace built in whole line completions with ones that use nvim-cmp.

The hotkey is wack, but it is default! So I think it makes the most sense, even if there could be other, more ergonomic mappings.

At the very least, if a person wants the functionality, but on a different mapping, they now have the option to take a peek at this community pack, copy paste it into their config, and change the mapping to what they want.

### CODE REVIEW

File: lua/astrocommunity/completion/nvim-cmp-buffer-lines/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# nvim-cmp-buffer-lines
+
+_Replaces_ the `<C-x><C-l>` whole line complete, to use nvim-cmp instead, using the following source:
```
Comment (Uzaaft) [inline lua/astrocommunity/completion/nvim-cmp-buffer-lines/README.md:3]: The standard is to use the "one-liner" from the github repo

```suggestion
nvim-cmp source for buffer words
```

File: lua/astrocommunity/completion/nvim-cmp-buffer-lines/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+**Repository:** <https://github.com/amarakon/nvim-cmp-buffer-lines>
```
Comment (Uzaaft) [inline lua/astrocommunity/completion/nvim-cmp-buffer-lines/README.md:5]:
```suggestion
**Repository:** <https://github.com/amarakon/nvim-cmp-buffer-lines>

- _Replaces_ the `<C-x><C-l>` whole line complete, to use nvim-cmp instead.
```

**Comment by Uzaaft** [comment]
Curious as to if you want to keep the note about <C-x><C-l> or not. Besides that, this looks good!

**Comment by Axlefublr** [comment]
You wanted me to remove it, no?

**Comment by Uzaaft** [comment]
> You wanted me to remove it, no?

I moved it to under the repository link 😅

**Comment by Uzaaft** [comment]
Nice. I was wondering why you weren't in the discord anymore. Seeing this PR from you made my day @Axlefublr

**Review by Uzaaft — APPROVED** [review]

**Pull request merged**